inc/admin/namespace.php

```diff
@@ -292,6 +292,10 @@ function render_new_page( WP_Post $post ) {
  * @return void
  */
 function handle_action( int $post_id ) {
+	if ( ! current_user_can( 'manage_options' ) ) {
```

Comment:
I'd suggest adding this to each of the action functions (`on_sync()`, etc.) so that even if implementations change later, those functions can never be called without the correct capability. That's also how `on_create()` is done above.